src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -329,15 +332,10 @@ private List<int> FindPeersToResync(ChainedHeader consensusTip)
                     ChainedHeader fork = this.FindForkIfChainedHeadersNotOnSameChain(peerTip, consensusTip);
 
                     // Do nothing in case peer's tip is on our consensus chain.
-                    if (fork != null)
+                    if ((fork != null) && (fork.Height <= this.finalizedBlockHeight.GetFinalizedBlockHeight()))
                     {
-                        int reorgLength = consensusTip.Height - fork.Height;
-
-                        if (reorgLength > maxReorgLength)
-                        {
-                            peerIdsToResync.Add(peerId);
-                            this.logger.LogTrace("Peer with Id {0} claims a chain that violates max reorg, its tip is '{1}'.", peerId, peerTip);
-                        }
+                        peerIdsToResync.Add(peerId);
+                        this.logger.LogTrace("Peer with Id {0} claims a chain that violates max reorg, its tip is '{1}'.", peerId, peerTip);
```

Comment:
It might be nice to have the actual finalise block height in the log as well for comparison.

Reply:
Fixed